tests/test_device/test_ipu/test_ipu_hooks.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import logging
+import os.path as osp
+import random
+import string
+import tempfile
+
+import pytest
+import torch
+import torch.nn as nn
+
+from mmcv.device.ipu import IS_IPU
+from mmcv.runner import build_runner
+from mmcv.runner.fp16_utils import auto_fp16
+
+if IS_IPU:
+    from mmcv.device.ipu.hook_wrapper import IPUFp16OptimizerHook
+
+skip_no_ipu = pytest.mark.skipif(
+    not IS_IPU, reason='test case under ipu environment')
+
+
+# TODO Once the model training and inference interfaces
+# of MMCLS and MMDET are unified,
+# construct the model according to the unified standards
+class ToyModel(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+        self.conv = nn.Conv2d(3, 3, 1)
+        self.bn = nn.BatchNorm2d(3)
+        self.relu = nn.ReLU6()
+        self.fp16_enabled = False
+
+    @auto_fp16(apply_to=('img', ))
+    def forward(self, img, return_loss=True, **kwargs):
+        x = self.conv(img)
+        x = self.bn(x)
+        x = self.relu(x)
+        if return_loss:
+            loss = ((x - kwargs['gt_label'])**2).sum()
+            return {
+                'loss': loss,
+                'loss_list': [loss, loss],
+                'loss_dict': {
+                    'loss1': loss
+                }
+            }
+        return x
+
+    def _parse_losses(self, losses):
+        return losses['loss'], losses['loss']
+
+    def train_step(self, data, optimizer=None, **kwargs):
+        losses = self(**data)
+        loss, log_vars = self._parse_losses(losses)
+        outputs = dict(
+            loss=loss, log_vars=log_vars, num_samples=len(data['img'].data))
+        return outputs
+
+
+@skip_no_ipu
+def test_optimizerhook():
+
+    model = ToyModel()
+    dummy_input = {
+        'data': {
+            'img': torch.rand((16, 3, 10, 10)),
+            'gt_label': torch.rand((16, 3, 10, 10))
+        }
+    }
+
+    temp_root = tempfile.gettempdir()
```

Comment:
This temporary directory should be removed after testing or use the tmp_path fixture from pytest (https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture)